### PR TITLE
[core] Support NumericPrimitiveToTimestamp in casting

### PR DESCRIPTION
**Pull request by xuzifu666**
### Purpose
Currently paimon casting not support numeric to timestamp cast，the pr is aimed to support it.

Linked issue: https://github.com/apache/paimon/issues/3833

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/casting/NumericPrimitiveToTimestamp.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        ZoneId zoneId =
+                targetType.is(DataTypeRoot.TIMESTAMP_WITH_LOCAL_TIME_ZONE)
+                        ? ZoneId.systemDefault()
+                        : DateTimeUtils.UTC_ZONE.toZoneId();
+        switch (inputType.getTypeRoot()) {
+            case INTEGER:
+            case BIGINT:
+                return value ->
+                        Timestamp.fromLocalDateTime(
+                                DateTimeUtils.toLocalDateTime(value.longValue(), zoneId));
```
[the remark this answers is not in the document]
Comment (xuzifu666) [reply paimon-core/src/main/java/org/apache/paimon/casting/NumericPrimitiveToTimestamp.java:57]: Yes，SQL Standard should be second unit，would modify it，Thanks~ @JingsongLi

File: paimon-core/src/test/java/org/apache/paimon/casting/CastExecutorTest.java [hunk 2]
```diff
@@ -101,6 +101,14 @@ public void testNumericToNumeric() {
                 CastExecutors.resolve(new FloatType(false), new DoubleType(false)), 1F, 1D);
     }
 
+    @Test
+    public void testNumericToTimestamp() {
+        compareCastResult(
```
[the remark this answers is not in the document]
Comment (xuzifu666) [reply paimon-core/src/test/java/org/apache/paimon/casting/CastExecutorTest.java:106]: OK，add it latter.

File: paimon-core/src/main/java/org/apache/paimon/casting/NumericPrimitiveToTimestamp.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (xuzifu666) [reply paimon-core/src/main/java/org/apache/paimon/casting/NumericPrimitiveToTimestamp.java]: Done，had change from mills to second

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**